src/utils/cloneSkillHelper/cloneSkillTemplate.ts

```diff
@@ -0,0 +1,116 @@
+import * as fs from 'fs';
+import * as fsExtra from 'fs-extra';
+import * as path from 'path';
+import * as R from 'ramda';
+import * as vscode from 'vscode';
+
+import { SKILL_FOLDER, BASE_STATES_CONFIG, BASE_RESOURCES_CONFIG } from '../../constants';
+import { AskError, loggableAskError } from '../../exceptions';
+import { Logger } from '../../logger';
+import { SkillInfo } from '../../models/types';
+import { CommandContext, SmapiResource, Utils } from '../../runtime';
+import { getSkillNameFromLocales } from '../skillHelper';
+import { openWorkspaceFolder } from '../workspaceHelper';
+
+export class CloneSkillTemplate {
```

Comment:
Need a better name for this class. Considering it is being extended and used across cloning both hosted and non-hosted skills, maybe call it the `abstractCloneSkill` class or refactor to a generic `CloneSkillHelper` class? `AbstractCloneSkill` makes more sense since you have a `cloneSkill` method in this class that throws an error when not implemented.

Reply:
Will use `AbstractCloneSkill `